### PR TITLE
New option testthat.summary.omit_dots to omit progress dots

### LINKED ISSUE #520: Option to stop tests earlier if more than n fail
Because it typically indicates something is seriously wrong, and you just want to see the errors right away.

### PR DESCRIPTION
**Pull request by krlmlr**
The dots account for roughly 10%-20% of the testing time, this is difficult to parallelize (#486). Default: Show dots.

Now also fixes #520.

### CODE REVIEW

**Comment by krlmlr** [comment]
Testing in RStudio seems to ignore .Rprofile, I don't know how to set this option globally so that it is also respected when testing in RStudio.

File: R/reporter-summary.R [hunk 1]
```diff
@@ -41,22 +43,23 @@ SummaryReporter <- R6::R6Class("SummaryReporter", inherit = Reporter,
     },
 
     add_result = function(context, test, result) {
+      dot <- NULL
```
Comment (hadley) [inline R/reporter-summary.R]: I don't particularly like the flow of logic here, but I can't see an obviously better approach.

File: tests/testthat/test-reporter.R [hunk 2]
```diff
@@ -65,7 +65,7 @@ test_that("reporters produce consistent output", {
     save_report("debug")
   )
   save_report("check", error_regexp = NULL)
-  save_report("summary", SummaryReporter$new(show_praise = FALSE))
+  save_report("summary", SummaryReporter$new(show_praise = FALSE, omit_dots = FALSE))
```
Comment (hadley) [inline tests/testthat/test-reporter.R]: I think we also need a test with `omit_dots = TRUE`

**Comment by hadley** [comment]
And a bullet for news please (I'll handle any merge conflicts)

**Comment by krlmlr** [comment]
@hadley: Rebased, added NEWS. Still needs a description in the help, waiting for #559.

File: tests/testthat/test-reporter.R (hunk 2 above)
Comment (krlmlr) [inline tests/testthat/test-reporter.R]: Added as "summary-no-dots".

**Comment by krlmlr** [comment]
Added help description, should be complete now.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**